### PR TITLE
[addons] fix: enable the chosen language add-on on startup if disabled

### LINKED ISSUE #18384: Kodi crashes on startup when Addons.db is removed and language pack is not default 'English'
### Describe the bug
When database file AddonsXY.db is removed prior to startup for any reason (testing, db-corruption etc.) and the installed and selected language pack addon is not the default `English` Kodi crashes.

The crash happens after the re-creating the database file.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
Kodi should recreate the missing Addons database and startup normally.

## Actual Behavior
db file is created but kodi crashes afterwards

### To Reproduce
1. set your gui language to anything but english (for example german, has alse been reproduced with french)
2. remove your database file(s) AddonsXY.db from directory `.kodi/userdata/Database`
3. start kodi

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/jurobicoya.kodi

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Debian Stretch, Ubuntu 18, Ubuntu 20
 - Kodi versions: current master built from source - leia from official ubuntu ppa

@da-anda could you please add information if you reproduced this on windows?

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by howie-f**
## Description

closes #18384 (startup issues with `addons.db` / language settings)

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

runtime tested and working

### CODE REVIEW

**Comment by ksooo** [comment]
While the fix seems correct, the commit message and PR title is not.

The fix is to *ignore* disabled resource language addons on startup and to boot Kodi with the always available default language addon.

Commit message and PR title should be corrected accordingly.

**Comment by howie-f** [comment]
> The fix is to _ignore_ disabled resource language addons on startup and to boot Kodi with the always available default language addon.

decided now to enable the language addon like @phunkyfish suggested yesterday.

File: xbmc/LangInfo.cpp [hunk 1]
```diff
@@ -661,17 +661,26 @@ bool CLangInfo::SetLanguage(std::string language /* = "" */, bool reloadServices
   if (language.empty())
     language = CServiceBroker::GetSettingsComponent()->GetSettings()->GetString(CSettings::SETTING_LOCALE_LANGUAGE);
 
+  auto& addonMgr = CServiceBroker::GetAddonMgr();
   ADDON::AddonPtr addon;
-  if (!CServiceBroker::GetAddonMgr().GetAddon(language, addon, ADDON::ADDON_RESOURCE_LANGUAGE, false))
+  if (!addonMgr.GetAddon(language, addon, ADDON::ADDON_RESOURCE_LANGUAGE, true))
```
Comment (phunkyfish) [inline xbmc/LangInfo.cpp:666]: I would add a comment here. `// Find the chosen language add-on if it's enabled`. Cause no one will know what the true refers to.

**Review by phunkyfish — APPROVED** [review]
Nice

**Comment by ksooo** [comment]
If it works, fine by me.

**Comment by howie-f** [comment]
yes, this is doing fine too. would have saved me a few hours yesterday. but oh well

File: xbmc/LangInfo.cpp [hunk 2]
```diff
@@ -661,17 +661,28 @@ bool CLangInfo::SetLanguage(std::string language /* = "" */, bool reloadServices
...
+                "CLangInfo: could not find or enable language add-on '%s', loading default..",
+                language.c_str());
+      language = std::static_pointer_cast<const CSettingString>(
+                     CServiceBroker::GetSettingsComponent()->GetSettings()->GetSetting(
+                         CSettings::SETTING_LOCALE_LANGUAGE))
+                     ->GetDefault();
+
+      if (!addonMgr.GetAddon(language, addon, ADDON::ADDON_RESOURCE_LANGUAGE, false))
+      {
+        CLog::Log(LOGFATAL, "CLangInfo: could not find default language add-on '%s'",
```
Comment (AlwinEsch) [inline xbmc/LangInfo.cpp:682]: Use as:
```cpp
CLog::Log(LOGFATAL, "CLangInfo: could not find default language add-on '{}'", language);
```

File: xbmc/LangInfo.cpp [hunk 3]
```diff
@@ -661,17 +661,28 @@ bool CLangInfo::SetLanguage(std::string language /* = "" */, bool reloadServices
...
+  // Find the chosen language add-on if it's enabled
+  if (!addonMgr.GetAddon(language, addon, ADDON::ADDON_RESOURCE_LANGUAGE, true))
+  {
+    if (addonMgr.IsAddonDisabled(addon->ID()) && !addonMgr.EnableAddon(addon->ID()))
     {
-      CLog::Log(LOGFATAL, "CLangInfo: could not find default language add-on '%s'", language.c_str());
-      return false;
+      CLog::Log(LOGWARNING,
+                "CLangInfo: could not find or enable language add-on '%s', loading default..",
+                language.c_str());
```
Comment (AlwinEsch) [inline xbmc/LangInfo.cpp:674]: Use as:
```cpp
      CLog::Log(LOGWARNING,
                "CLangInfo: could not find or enable language add-on {}', loading default",
                language);
```

**Review by AlwinEsch — COMMENTED** [review]
It's about time that I tackle and revise the AddonManager. Always having to use GetAddon for almost everything I find extreme.

For here at the request first OK.

File: xbmc/LangInfo.cpp (hunk 2 above)
Comment (howie-f) [reply xbmc/LangInfo.cpp:682]: oh, yes the new logging conventions. sure will come in

**Comment by howie-f** [comment]
jenkins build this please

**Review by AlwinEsch — APPROVED** [review]

**Comment by AlwinEsch** [comment]
Just a general note, maybe good to add a User related note to requests. We still not have now proper regulations about, I'm find it only good to have a small User understandable text on requests, where can be maybe taken for a new release changelog :thinking: 

Here e.g. how my one looks now from last 2 months :grinning: (and need to look about older dates :smirk:):

---------------------------

Kodi Changelog:

Video scans:
-------------------
- Fix: Content recognition (e.g. videos) of data from a packed archive (e.g. * .rar)
  - Previously, the data stored in the archive was not recognized correctly in some cases and was not shown in Kodi's movies and TV shows.
  - https://github.com/xbmc/xbmc/pull/18210

Music:
-------------------
- Fix: List view of music files which contain multiple streams and are used as folders.
  - Previously, the necessary information was not set in its content and e.g. artist and title name not available.
   > This primarily refers to files which are processed by audiodecoder addons.
- Feature: Audio decoder addons can now pass more information on Kodi (album, year, notes ...) 
  - Before was only the title, artist and duration possible to give Kodi

Addon system:
-------------------
- Improvement: Asks you if you have an add-on marked as broken in your system and want to activate it
- Improvement: New GUI list items added

  | Name | Description
  |------|------------
  | ListItem.AddonLifecycleType | To get related type string where also can be "broken", "deprecated" and "normal"
  | ListItem.AddonLifecycleDesc | To get the text where was before on "ListItem.AddonBroken"
  > *NOTE:* `ListItem.AddonBroken` deprecated, but still supported
  - New json webinterface **shema** type for `deprecated`within `Addon.Details` and `Addon.Fields`

skin.estuary:
-------------------
- Improvement: Shows broken declared in addon list using addon picture and list icon

Binary addon system:
-------------------
- Fix: Prevent VFS addon changes (update, disable and uninstall) if in use
- Fix: VFS addon problem, where Kodi has needed restart before to show new installed version
- Improvement: Binary addon system revised to ensure higher security in its data exchange between addon and Kodi
- Improvement: Revised binary addon system documentation for outside developers
- Improvement: Make internal binary addon settings with help texts and cleaner
  - audioencoder.kodi.builtin.aac
  - audioencoder.kodi.builtin.wma
  - screensaver.xbmc.builtin.dim

**Comment by phunkyfish** [comment]
You have 2 approvals, your button @howie-f.

**Comment by howie-f** [comment]
thx 2 @ksooo & @phunkyfish :+1:

**Pull request merged**